packages/gitbook/src/lib/logger.ts

```diff
@@ -0,0 +1,60 @@
+export type LogLabelValue = string | number | undefined | boolean;
+
+export type LogLabels = { [key: string]: LogLabelValue };
+
+export type SubLoggerOptions = {
+    labels?: LogLabels;
+};
+
+export type LoggerOptions = SubLoggerOptions & {
+    requestId?: string;
+};
+
+const formatPrefix = (name: string, options: LoggerOptions) => {
+    const { labels } = options;
+    return `%${JSON.stringify(labels ?? {})}%[${name}]`;
```

Comment:
We should comment to explain that this what we have in our tail worker to extract labels.